utcp.c

```diff
@@ -1864,7 +1866,7 @@ struct timeval utcp_timeout(struct utcp *utcp) {
             if(c->reapable) {
                 debug("Reaping %p\n", c);
                 free_connection(c);
-                i--;
+                --i;
```

Comment:
`next_conn` needs to be decremented here as well, otherwise a connection will be skipped.